icechunk/src/format/snapshot.rs

```diff
@@ -49,6 +49,55 @@ pub struct ZarrArrayMetadata {
     pub dimension_names: Option<DimensionNames>,
 }
 
+impl ZarrArrayMetadata {
+
+    /// Returns an iterator over the maximum permitted chunk indices for the array.
+    ///
+    /// This function calculates the maximum chunk indices based on the shape of the array
+    /// and the chunk shape.
+    ///
+    /// # Returns
+    ///
+    /// An iterator over the maximum permitted chunk indices.
+    fn max_chunk_indices_permitted(&self) -> impl Iterator<Item = u64> + '_ {
+        self.shape
+            .iter()
+            .zip(self.chunk_shape.0.iter())
+            .map(|(s, cs)| ((s + cs.get() - 1)) / cs.get() - 1)
+    }
+
+    /// Validates the provided chunk coordinates for the array.
+    ///
+    /// This function checks if the provided chunk indices are valid for the array.
+    ///
+    /// # Arguments
+    ///
+    /// * `coord` - The chunk indices to validate.
+    ///
+    /// # Returns
+    ///
+    /// An `IcechunkResult` indicating whether the chunk coordinates are valid.
+    ///
+    /// # Errors
+    ///
+    /// Returns `IcechunkFormatError::ChunkCoordinatesNotFound` if the chunk coordinates are invalid.
+    pub fn valid_chunk_coord(&self, coord: &ChunkIndices) -> IcechunkResult<bool> {
```

Comment:
Actually, this function doesn't really fail. Things would be easier if you make it return a simple `bool`